### ISSUE TITLE
Add UI to filter on a value being NULL

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#789]
## Problem

User has no way to add a filter condition specifying that a value should be NULL.

## Proposed solution

In addition to "Equals" and "Not Equals", we should have an "Is NULL" option when setting a filter condition.

**Comment by pavish** [comment]
@seancolsen The filters are meant to be dynamic i.e anything related to column types is meant to be dynamic.

For eg., for the type Number, the server should provide what kind of filters are applicable for number, what input type should be shown within the table, the grouping options we should allow for it, the icons required for it etc., This way we can easily add and remove types without having to update any frontend code, making it possible to provide data types as plugins.

I'm not sure how well we'd be able to get it done for the alpha release. But it's part of the frontend stories you could see in https://github.com/centerofci/mathesar/milestone/11 and an additional issue that tracks some of it: https://github.com/centerofci/mathesar/issues/661

This is to say that the "Equals" and "Not Equals" are both temporary conditions that were hardcoded on the frontend, and they would be eventually provided by the server, which will include isNull.

@kgodey I assume this is our plan for the alpha. Please let us know if you think it would be better to hardcode type specific properties on the frontend instead.

**Comment by kgodey** [comment]
@pavish I think you should hardcode them for now and we can remove the hardcoding once #661 is completed. That way, we have the freedom to put off #661 until after the alpha release if needed.

**Comment by pavish** [comment]
@kgodey Thanks. Sounds good to me.

**Comment by pavish** [comment]
@seancolsen I think this can wait until type configuration is complete as part of #500.

**Comment by pavish** [comment]
This issue also depends on the new filters structure.

Blocked by #1001 and #500

**Comment by pavish** [comment]
Closing this issue since it's no longer relevant. The filters are based on each column type and they contain the filter `is empty` which handles this usecase.
